Remove mp32

gferraro opened the pull request.

arthurmcgregor commented:
Hmmmm so its failing the travis-ci check because the next variable is not defined. To pass it along from the route I took the same approach to what you have done with the api variable. However, I defined next in login.pug directly (whereas api is defined in base.pug)

So I'm not sure why next is failing the check when, in the same file a few lines earlier, api doesn't cause it to fail....

CameronRP commented:
The api variable passed because of this line added to the top of user.js `/* global api, Promise */`
You should be able to add the line `/* global next, Promise */` to fix that issue.

arthurmcgregor commented:
That sorted it - thanks :-)

mjs commented on public/javascripts/includes/user.js:
```diff
@@ -160,3 +164,20 @@ user.getHeaders = function() {
+// Do not run the loginRedirect on the public facing web pages
+if (window.location.pathname.slice(0,6) !== '/login' && window.location.pathname !== '/register' && window.location.pathname !== '/' ) {
+  loginRedirect();
+}
```
A few things:
- why use `slice()` just for `/login`?
- does the next URL need to be URI encoded?
- why do you remove the initial `/`?

Also, something like this is perhaps a little clearer (untested!):

```js

// These paths don't require that the user is logged in.
const noAuthPaths = Object.freeze([
  '/login', 
  '/register', 
  '/',
])

function loginRedirect(pathname) {
  if (noAuthPaths.includes(pathname) {
    return;
  }
  if (!user.isLoggedIn()) {
    window.location.href = '/login' + '?next=' + encodeURIComponent(pathname);
  }
}

loginRedirect(window.location.pathname);
```

arthurmcgregor commented:
Thanks for the feedback @mjs. Have implemented your suggestions. The slicing was a hangover from before I realised there was an inbuilt process for handling query strings and wasn't needed anymore. URI encoding probably a good idea.

hardiesoft committed:
- Porting CptvPlayer to Vue3 - WIP

hardiesoft committed:
- Porting CptvPlayer to Vue3 - WIP (2)

hardiesoft committed:
- Porting CptvPlayer to Vue3 - WIP (3)

hardiesoft committed:
- Porting CptvPlayer to Vue3 - WIP (4)

hardiesoft committed:
- Porting CptvPlayer to Vue3 - WIP (5)
- Fixing edge case with back-dating station creation.

hardiesoft committed:
Merge remote-tracking branch 'origin/heirarchical-tags' into browse-next-ui-3

hardiesoft committed:
- Porting CptvPlayer to Vue3 - WIP (6)
- Fix release

Commit:
allow tracker version to be a string

hardiesoft committed:
- Making initial email confirmation setup work properly.

hardiesoft committed:
Merge remote-tracking branch 'origin/main' into browse-next-ui-3

hardiesoft committed:
- Configuring eslint+prettier for browse-next

hardiesoft committed:
- Configuring eslint+prettier for browse-next

hardiesoft committed:
- Linting and updating deps

hardiesoft committed:
- Linting and updating deps (2)

hardiesoft committed:
- Linting and updating deps (2)

hardiesoft committed:
- Fix tests

Commit:
Merge branch 'main' into tracker-string

Commit:
Merge remote-tracking branch 'origin/tracker-string' into browse-next-ui-3

hardiesoft committed:
- Add ability to create alerts for stations
- Add ability to delete alerts
- Add ability to get alerts for stations
- Removed redundant Device and User includes from alerts returned from API.
- Add ability to get alerts for the current user
- Make alerts aware of hierarchy, so that you can ask for alerts on whole categories
- UI to manage alerts in browse

hardiesoft committed:
Merge remote-tracking branch 'origin/heirarchical-tags' into browse-next-ui-3

hardiesoft committed:
Merge branch 'browse-next-ui-3' of https://github.com/TheCacophonyProject/cacophony-web into browse-next-ui-3

hardiesoft committed:
- Species alerts only trigger for device uploads
- Recordings that get species alerts have to be less than 24hrs old
- Added and amended alerts tests.

hardiesoft committed:
- Fix migration teardown.

hardiesoft committed:
- Try to fix eslint error.

hardiesoft committed:
- Try to fix eslint error (2)

hardiesoft committed:
- Try to fix eslint error (3)

hardiesoft committed:
- Try to fix eslint error (4)

hardiesoft committed:
- Try to fix eslint error (5)

hardiesoft committed:
- Try to fix eslint error (7)

hardiesoft committed:
- Allow any tags for alerts, restrict alert length.

hardiesoft committed:
- Add prompt for alert removal, link in emails.
- Fix false positives in alerts.
- Change alert since display.

hardiesoft committed:
Merge branch 'main' of https://github.com/TheCacophonyProject/cacophony-web into browse-next-ui-3

hardiesoft committed:
- Fix typo in test.

hardiesoft committed:
- Lint tests

hardiesoft committed:
- Fix remove alert link in email

hardiesoft committed:
- Fix remove alert link in email (2)

hardiesoft committed:
- Allow bird tag alerts
- Remove alert name from UI.

Zainrax committed:
feat: add bulk delete api & UI

This adds two endpoints to recordings, /delete & /undelete. Delete takes in a normal recording query and soft deletes recordings found. Undelete takes in a list of ids to undelete.

Zainrax committed:
Merge tag 'v1.9.26' into heirarchical-tags

Zainrax committed:
fix(api): countAll query

hardiesoft committed:
- dashboard polishing for browse-next

hardiesoft committed:
- make side-nav stick open on wide screens.

Zainrax committed:
fix(bulk-delete): add proper count, refactor /count {get} endpoint

Zainrax committed:
fix(bulk-delete): add group admin check on count

Zainrax committed:
fix: lint

Zainrax committed:
fix(count): add deleted query

Zainrax committed:
fix(count): move deleted to where

Zainrax committed:
fix(bulk-delete): optimize update, remove limit

hardiesoft committed:
- Fix getOneForRecording to bias to recordings that have station alerts.

hardiesoft committed:
- Fix getOneForRecording to bias to recordings that have station alerts.

Zainrax committed:
fix(bulk-delete): revert query, limit 1000, change text

hardiesoft committed:
- More WIP mobile responsive css work for browse-next

hardiesoft committed:
- More WIP mobile responsive css work for browse-next, integrating hierarchical tagging.

Zainrax committed:
test(bulk-delete): check delete & undelete for both member & non-member

Zainrax committed:
test: restore recording add

Zainrax committed:
fix: remove dev env changes

Zainrax committed:
Merge tag 'v1.9.28' into heirarchical-tags

hardiesoft committed:
- Tidy up recording tags table - remove redundant 'what' column.
- WIP responsive tagging interface.

hardiesoft committed:
- Adjust recording tag tests

hardiesoft committed:
- Adjust recording tag tests

hardiesoft committed:
Merge branch 'heirarchical-tags' of https://github.com/TheCacophonyProject/cacophony-web into browse-next-ui-3

 Conflicts:
	api/models/Recording.ts

hardiesoft committed:
- Linting

hardiesoft committed:
- Linting (2)

hardiesoft committed:
Merge branch 'main' of https://github.com/TheCacophonyProject/cacophony-web into browse-next-ui-3

 Conflicts:
	api/api/V1/AuthenticateUser.ts
	api/emails/transactionalEmails.ts
	api/models/Recording.ts
	browse-next/package-lock.json
	browse-next/package.json
	browse-next/src/App.vue
	browse-next/src/api/fetch.ts
	browse-next/src/components/RecordingViewLabels.vue
	browse-next/src/components/RecordingViewTracks.vue
	browse-next/src/components/TrackTaggerRow.vue
	browse-next/src/components/TracksScrubber.vue
	browse-next/src/components/VisitsDailyBreakdown.vue
	browse-next/src/components/cptv-player/CptvPlayer.vue
	browse-next/src/components/cptv-player/cptv-decoder/decoder.ts
	browse-next/src/components/cptv-player/cptv-decoder/decoder.worker.ts
	browse-next/src/font-awesome-icons.ts
	browse-next/src/models/LoggedInUser.ts
	browse-next/src/views/RecordingView.vue

Commit:
New thumbs (#58)

* new thumbs

* added track thumbnail tests, fixed a few bugs with thumbnails where meta not present

hardiesoft committed:
- Browse next tagging UI WIP

hardiesoft committed:
Merge branch 'browse-next-ui-3' of https://github.com/TheCacophonyProject/cacophony-web into browse-next-ui-3

hardiesoft committed:
- Browse next tagging UI WIP

hardiesoft committed:
- Browse next tagging UI WIP (2)

hardiesoft committed:
- Browse next tagging UI WIP (3)

hardiesoft committed:
- Browse next tagging UI WIP - Adding recording labels.

hardiesoft committed:
- Browse next tagging UI WIP - Adding recording labels.
- Adding quick flag/star buttons

hardiesoft committed:
- upgrade travis dist

hardiesoft committed:
- revert travis dist and set node version to 17

hardiesoft committed:
- set node version to 16

hardiesoft committed:
- Tweaking custom tag adding UI

hardiesoft committed:
- Improved the visits calculation to not prefer user false-positive tags over user animal tags, and to split visits where there are two user animal tags in the same time period across multiple recordings making up a visit.  I've also fixed the multiple animals problem for user tags, so if you tag two different animals in a single recording, it will be split into two visits, one for each tag, each containing the same recording.
- We will now prefer an animal tag in the event of an AI animal tag amongst lots of false-positives.  We are reasonably confident that the number of false false-postives is fairly low (< 20%).
- Renamed "unknown" to unidentified
- Made pinned tags be specific to a user per group.

hardiesoft committed:
- Bump classification version.

hardiesoft committed:
- Update path in TrackTags for unknown->unidentified tags

hardiesoft committed:
- Update browse UI for only showing "unidentified", not "unknown"

hardiesoft committed:
- Fix group user settings api endpoint

hardiesoft committed:
- Fix group user settings api endpoint (2)

hardiesoft committed:
- Fixes

hardiesoft committed:
- Fixes (2)

hardiesoft committed:
- Fixes (3)

hardiesoft committed:
- Fixes (4)

hardiesoft committed:
- Fixes (5)

hardiesoft committed:
- Make sure group user settings are included with user groups (user pinned tags per group).
- Make user level pinning update correctly.
- Prefer AI tags of animals/things over user tags of false-positive when calculating visits.

hardiesoft committed:
- Always show other user tags for tracks if any.
- Add tie-breaking logic to visits calculations for visits that only have AI tags.

hardiesoft committed:
- Always show other user tags for tracks if any.
- Add tie-breaking logic to visits calculations for visits that only have AI tags.

hardiesoft committed:
- Fix tests

hardiesoft committed:
- Recalculate visits when user tags change.

hardiesoft committed:
- Linting

hardiesoft committed:
- Update current visit correctly

hardiesoft committed:
- Port export mp4, advanced export, download CPTV and delete recording functionality (WIP).

hardiesoft committed:
- Port functionality to show CPTV header info.

hardiesoft committed:
- Fix release build

hardiesoft committed:
- Add loading spinners to tracks + labels panels

hardiesoft committed:
- Add slenderbilled whiteeye to classifications.json
- Add API + test for adding a classifier event with an associated thumbnail, and for retrieving it.

hardiesoft committed:
- Add slenderbilled whiteeye to classifications.json
- Add API + test for adding a classifier event with an associated thumbnail, and for retrieving it.

hardiesoft committed:
- Api docs

hardiesoft committed:
- Linting

hardiesoft committed:
- Reorder API calls for events from less general to more general.

hardiesoft committed:
- Add thumbnail to fixtures

hardiesoft committed:
- Fallback to clip thumbnail if thumbnail for track doesn't exist.

hardiesoft committed:
- Fix background toggle on CPTVPlayer
- Show device name next to station name in recording detail.
- Add UI to manage group-level tag preferences.

hardiesoft committed:
- WIP user management/user-invite UI.

hardiesoft committed:
- Add logging for server errors where the payload is too large to figure out which device is sending too large requests.

hardiesoft committed:
- Add logging for server errors where the payload is too large to figure out which device is sending too large requests.

hardiesoft committed:
- Make sure we're correctly tracking rawFileSizes and fileSizes

hardiesoft committed:
- Make sure we're correctly tracking rawFileSizes and fileSizes (2)

hardiesoft committed:
- Make sure we return scheduleIds for devices that have schedules.

hardiesoft committed:
- Make sure we return scheduleIds for devices that have schedules. (2)

hardiesoft committed:
- Flatten error log output

hardiesoft committed:
- Increase JSON payload size to 2MB

hardiesoft committed:
- Add usage tracking fields and associated book-keeping to enable billing.

hardiesoft committed:
- Add usage tracking fields and associated book-keeping to enable billing (2)

hardiesoft committed:
- Add chicken tag
- Add user removedAt instead of removing group users (for billing purposes)
- Flesh out user invitation APIs, add GroupInvites table, pending group users (for existing users).
- Removed undefined rows from Schedules

hardiesoft committed:
- Add chicken tag
- Add user removedAt instead of removing group users (for billing purposes)
- Flesh out user invitation APIs, add GroupInvites table, pending group users (for existing users).
- Removed undefined rows from Schedules

hardiesoft committed:
- WIP adding tests for group user invitations.

hardiesoft committed:
- Adding tests for transactional emails.

hardiesoft committed:
- Finish adding tests for transactional emails.

hardiesoft committed:
- Add a message to browse about browse-next
- Start tidying up views of browse-next for preview release.
- Add settings to DeviceHistory entries.

hardiesoft committed:
- Add a message to browse about browse-next
- Start tidying up views of browse-next for preview release.
- Add settings to DeviceHistory entries.

hardiesoft committed:
- UI Fixes

hardiesoft committed:
- UI Fixes

hardiesoft committed:
- UI Fixes (2)

hardiesoft committed:
- Table refactors

hardiesoft committed:
- Edit user permissions modal

hardiesoft committed:
- Edit user permissions modal
- Fix tests

hardiesoft committed:
- Rework tests that upload recordings to be more parallel and less slow.

hardiesoft committed:
- Refactoring card table, making card views for user management at smaller breakpoints

hardiesoft committed:
- Adding browser e2e tests for browse-next signup and login + group membership flows.

hardiesoft committed:
- Adding browser e2e tests for browse-next signup and login + group membership flows (2)

hardiesoft committed:
- More e2e tests and fixes around user login/password change flows.

hardiesoft committed:
- Fix tests

hardiesoft committed:
- Fix tests + Don't run front-end e2e tests in travis

hardiesoft committed:
- Add favicons, page titles, more unimplemented modals.

hardiesoft committed:
- Add favicons, page titles, more unimplemented modals (2)

hardiesoft committed:
- Add favicons, page titles, more unimplemented modals (3)

hardiesoft committed:
- Starting to add some icons

hardiesoft committed:
- Tag icons can be resolved hierarchically, added some more preliminary tag icons.

hardiesoft committed:
- Extend email timeout for intermittently failing tests,

hardiesoft committed:
- Fix delete button, update docker-base

hardiesoft committed:
- Fix delete button, update docker-base (2)

hardiesoft committed:
- Fix decoder re-init

hardiesoft committed:
- Fix export pixel ratio on retina screens

hardiesoft committed:
- Check for removed users in groups

hardiesoft committed:
- Fix invitation link routing, and write tests for it.
- Change the way playback and tag expansion happens when you select a track.

hardiesoft committed:
Merge branch 'main' of https://github.com/TheCacophonyProject/cacophony-web into browse-next-ui-3

# Conflicts:
#	api/classifications/classification.json
#	api/classifications/label_paths.json
#	api/emails/transactionalEmails.ts

hardiesoft committed:
- doc trailcam images wip

hardiesoft committed:
- doc trailcam images wip (2)

hardiesoft committed:
- doc trailcam images wip (3)
- Allow activity search, so uploaded results can be seen.

hardiesoft committed:
Merge branch 'main' of https://github.com/TheCacophonyProject/cacophony-web into doc-t2m-wip

jwr87 committed:
Added analysis tab in group view with inital layout and CI index by device working

jwr87 committed:
Added function to browse to allow the device cacophony index endpoint to be accessed; also fixed a bug with said endpoint (sequelize query was hanging due to naming error

hardiesoft committed:
- activity search with recordings view.

jwr87 committed:
Fixed formatting error in getDeviceCacophonyIndex func, implemented first draft of cacophony index over time visual (currently using fake data as req endpoint broken)

hardiesoft committed:
- Fix conflicting tags evaluation to work with tag hierarchies

hardiesoft committed:
- Adding visits mode query to activity search.

jwr87 committed:
Seperated index time chart into its own component, did some tidying up around this - added custom date range

jwr87 committed:
Reworked dates to be shared across graph while also adding baseline for device vs station option + bug fix in device api

jwr87 committed:
Got station / device switch working so can now swap between two (includes adding station cacophony index endpoint)

jwr87 committed:
Added species-counts endpoints to retrieve number of identifications per species in range

jwr87 committed:
Added functionality for percentage of identified species per device or station inc. req endpoints

jwr87 committed:
Added a bulk endpoint for cacophony index as well as added ability to select multiple devices/stations in the species over time graph

jwr87 committed:
Reworked cacophony index over time to use bulk queries and added station endpoints and frontend reqs for species over time by stations

jwr87 committed:
Small refactor and added video tab for visuals + backend for days active endpoint and frontend for average daily species visits from videos per device (not station yet)

jwr87 committed:
Added getDaysActive to station model

Jackwryanz committed:
Apply Prettier formatting to project files

Jackwryanz committed:
Further prettier cleanup

Jackwryanz committed:
Hopefully final prettier cleanup

Jackwryanz committed:
Api prettier fixining

Jackwryanz committed:
More api linting cleanup

Jackwryanz committed:
Removed strat '/>' and corrected device type filtering per page

Jackwryanz committed:
Modified date ranges and added auto interval based on date range selected

Jackwryanz committed:
Changed style of species of time to non stacked. No longer displaying devices with no data

Jackwryanz committed:
Linter cleanup

hardiesoft committed:
- Progress after Doc t2m milestone meeting.

hardiesoft committed:
- Linting

hardiesoft committed:
- Fix update of classifications client-side when tagging to handle false-positive visits properly.
If there are any animal human tags for a visit, prefer them over any amount of non-animal human tags.

hardiesoft committed:
- Break out prev/next visits into separate buttons on desktop views, add hotkeys for efficient tagging.
- Make hierarchical tag selector scroll into view where needed
- Make raw/archival downloads work for trailcam images

hardiesoft committed:
- Implement ability to show device/location at recording time reference photos in cptv player view.

hardiesoft committed:
- Pull additionalMetadata from trailcam images + OCR overlay footers to the best of our ability with the knowledge of the models we have.

hardiesoft committed:
- Allow unknown trailcams

hardiesoft committed:
- Make release build work for browse-next

hardiesoft committed:
- Fix migration rollback

hardiesoft committed:
- Fix migration rollback (2)

hardiesoft committed:
- Fix migration rollback (3)

hardiesoft committed:
Merge branch 'index-visualizations' of https://github.com/TheCacophonyProject/cacophony-web into doc-t2m-wip

# Conflicts:
#	api/api/V1/Device.ts

hardiesoft committed:
- Merge Jacks' changes

hardiesoft committed:
- Fix migration rollback

hardiesoft committed:
- Fix migration rollback (2)

hardiesoft committed:
- Refactor API to update to:
  - Ubuntu 22.04
  - Postgres 14
  - python3
  - Updated node module resolution to use esm modules, and resolved circular dependencies
  - Fix tests
  - Properly return responses early on duplicate recordings.

hardiesoft committed:
- Linting

hardiesoft committed:
- Updated to node 18, also for travis env.

hardiesoft committed:
- Use updated base docker image.

hardiesoft committed:
- Update to Typescript 5, fix scripts.

hardiesoft committed:
- Linting

hardiesoft committed:
- Make waitForApi be a module.

hardiesoft committed:
- Fix node_modules path for travis

hardiesoft committed:
- Optimising docker files

hardiesoft committed:
- Move base docker image creation into bash script to reduce layers and image size.

hardiesoft committed:
- Move base docker image creation into bash script to reduce layers and image size. (2)

hardiesoft committed:
- Optimising docker files (2)

hardiesoft committed:
- Move base docker image creation into bash script to reduce layers and image size. (3)

hardiesoft committed:
- Optimising docker files (3)

Commit:
resolve

Commit:
package updated

Commit:
fixed tests

Commit:
Merge remote-tracking branch 'origin/doc-t2m-wip' into remove-mp32
